frontend/packages/kubevirt-plugin/src/components/vm-disks/vm-disks.tsx

```diff
@@ -10,17 +10,31 @@ import { sortable } from '@patternfly/react-table';
 import { DataVolumeModel } from '../../models';
 import { VMGenericLikeEntityKind } from '../../types/vmLike';
 import { VMLikeEntityTabProps } from '../vms/types';
-import { getResource } from '../../utils';
+import { getResource, getLoadedData } from '../../utils';
 import { wrapWithProgress } from '../../utils/utils';
 import { diskModalEnhanced } from '../modals/disk-modal/disk-modal-enhanced';
-import { CombinedDiskFactory } from '../../k8s/wrapper/vm/combined-disk';
+import { CombinedDiskFactory, CombinedDisk } from '../../k8s/wrapper/vm/combined-disk';
 import { V1alpha1DataVolume } from '../../types/vm/disk/V1alpha1DataVolume';
 import { VHW_TYPES } from '../create-vm-wizard/tabs/virtual-hardware-tab/types';
 import { StorageBundle } from './types';
 import { DiskRow } from './disk-row';
 import { diskTableColumnClasses } from './utils';
 import { isVMI } from '../../selectors/vm';
 
+const lookupPVCName = (pvcs: FirehoseResult<K8sResourceKind[]>, disk: CombinedDisk) => {
```

Comment:
this logic should be moved to `CombinedDiskFactory#constructor` as all the lookup are executed and relevant data bundled there 

- this also takes into account loading of the pvcs and showing loading animation for that

Reply:
done